applications/ConstitutiveLawsApplication/python_scripts/symbolic_generation/iso_damage_tangent_tensor/isotropic_damage_automatic_differentiation.py

```diff
@@ -0,0 +1,351 @@
+def ComputePredictorStressVector3D(Young, nu, Strain0, Strain1, Strain2, Strain3, Strain4, Strain5):
+    c1 = Young / ((1.0 + nu) * (1.0 - 2.0 * nu))
+    c2 = c1 * (1.0 - nu)
+    c3 = c1 * nu
+    c4 = c1 * 0.5 * (1.0 - 2.0 * nu)
+
+    S0 = c2 * Strain0 + c3 * Strain1 + c3 * Strain2
+    S1 = c3 * Strain0 + c2 * Strain1 + c3 * Strain2
+    S2 = c3 * Strain0 + c3 * Strain1 + c2 * Strain2
+    S3 = c4 * Strain3
+    S4 = c4 * Strain4
+    S5 = c4 * Strain5
+
+    return S0, S1, S2, S3, S4, S5
+
+def ComputeJ2Invariant2D(Deviator, pmean):
+    return 0.5 * (Deviator[0]**2.0 + Deviator[1]**2.0 + pmean**2.0) + Deviator[2]**2.0
+def ComputeJ2Invariant3D(Deviator, pmean):
+    return 0.5*(Deviator[0]**2+Deviator[1]**2+Deviator[2]**2) + (Deviator[3]**2+Deviator[4]**2+Deviator[5]**2)
+
+def ComputeJ3_2D(Deviator):
+    return Deviator[0] * Deviator[1] - (Deviator[2]**2)
+def ComputeJ3_3D(Deviator):
+    return Deviator[0] * (Deviator[1] * Deviator[2] - Deviator[4] * Deviator[4]) + Deviator[3] * (-Deviator[3] * Deviator[2] + Deviator[5] * Deviator[4]) + Deviator[5] * (Deviator[3] * Deviator[4] - Deviator[5] * Deviator[1])
+
+def ComputePmean2D(StressVector):
+    return (StressVector[0] + StressVector[1]) / 3.0
+def ComputePmean3D(StressVector):
+    return (StressVector[0] + StressVector[1] + StressVector[2]) / 3.0
+
+def ComputeDamage(UniaxialStress, Threshold, Gf, L, Young, Softening):
+    if (Softening ==  "Linear"):
+        A = ComputeAParameterLinear(Threshold, Gf, L, Young)
+        return ComputeDamageLinear(UniaxialStress, Threshold, A)
+    else: # Exponential
+        A = ComputeAParameterExponential(Threshold, Gf, L, Young)
+        return ComputeDamageExponential(UniaxialStress, Threshold, A)
+
+def ComputeDamageLinear(UniaxialStress, Threshold, A):
+    return (1.0 - Threshold / UniaxialStress) / (1.0 + A)
+
+def ComputeDamageExponential(UniaxialStress, Threshold, A):
+    return 1.0 - (Threshold / UniaxialStress) * exp(A * (1.0 - UniaxialStress / Threshold))
+
+def ComputeAParameterLinear(Threshold, Gf, L, Young):
+    return -Threshold**2 / (2.0 * Young * Gf / L)
+
+def ComputeAParameterExponential(Threshold, Gf, L, Young):
+    return 1.0 / (Gf * Young / (L * Threshold**2) - 0.5)
+
+
+
+"""
+Here we define the problem in terms of dimension, yield surface and linear/exponential softening
+"""
+
+# INPUT DATA
+"""
+---------------------------------------------------------------------------------------------------
+"""
+dimension = 3 # 2, 3
+yield_surface = "DruckerPrager" # DruckerPrager, ModifiedMohrCoulomb, Rankine, VonMises
+softening = "Exponential" # Exponential, Linear
+"""
+---------------------------------------------------------------------------------------------------
+"""
+
+# Common variables
+mode = "c"
+
+J2 = 0.0; uniaxial_stress = 0.0; A = 0.0; damage = 0.0;
+if (dimension == 2):
+    Strain0, Strain1, Strain2, Seff, Stress, Deviator, Ct, Young, nu, threshold, Gf, characteristic_length = SetUp2DProblem()
```

Comment:
```suggestion
    Strain0, Strain1, Strain2, Deviator, Ct, Young, nu, threshold, Gf, characteristic_length = SetUp2DProblem()
```
I think you don't need to define the stresses a vector of symbols. What you're actually making is overwriting these with the components of a vector which are a function of the `Strain` symbols.